Rounding error in local_eulerian_grid_support_of_lagrangian_grid_kernel

I think the floor division done [here](https://github.com/SophT-Team/SophT-Simulator/blob/a281b4b75716c641020af3fe64b7c7ab2e5072b8/sopht/numeric/immersed_boundary_ops/EulerianLagrangianGridCommunicator2D.py#L117) might be susceptible to rounding error. This can happen silently and should probably be mitigated if we can.

In order to reproduce this issue, we can run the current [test case](https://github.com/SophT-Team/SophT-Simulator/blob/main/tests/test_numeric/test_immersed_boundary_ops/test_eulerian_lagrangian_grid_communicator_2d.py) by employing `n_values=24`. This should result in the error below:
```
>       np.testing.assert_allclose(
            ref_nearest_eul_grid_index_to_lag_grid,
            nearest_eul_grid_index_to_lag_grid,
            atol=get_test_tol(precision),
        )
E       AssertionError:
E       Not equal to tolerance rtol=1e-07, atol=2.22045e-13
E
E       Mismatched elements: 2 / 6 (33.3%)
E       Max absolute difference: 1
E       Max relative difference: 0.11111111
E        x: array([[ 8,  9, 10],
E              [ 8,  9, 10]])
E        y: array([[8, 9, 9],
E              [8, 9, 9]])

tests/test_numeric/test_immersed_boundary_ops/test_eulerian_lagrangian_grid_communicator_2d.py:98: AssertionError
```

As we can see, some of the indices are off by 1, due to rounding error. I'm not sure how we can mitigate this yet, especially given the kernel is a numba kernel too. Thoughts on this @bhosale2 ?

--- a/sopht/numeric/immersed_boundary_ops/EulerianLagrangianGridCommunicator2D.py
+++ b/sopht/numeric/immersed_boundary_ops/EulerianLagrangianGridCommunicator2D.py
@@ -114,9 +114,9 @@ def generate_local_eulerian_grid_support_of_lagrangian_grid_kernel_2d(
 
         """
         # dtype of nearest_grid_index takes care of type casting to int
-        nearest_eul_grid_index_to_lag_grid[...] = (
-            lag_positions - eul_grid_coord_shift
-        ) // dx
+        nearest_eul_grid_index_to_lag_grid[...] = np.rint(
+            (lag_positions - eul_grid_coord_shift) / dx
+        )
         # TODO We need to add boundary exception handling! where the Lagrangian
         #  node goes in `interp_kernel_width` boundary zone of the Eulerian grid
         # get relative distance (support) of body

--- a/sopht/numeric/immersed_boundary_ops/EulerianLagrangianGridCommunicator3D.py
+++ b/sopht/numeric/immersed_boundary_ops/EulerianLagrangianGridCommunicator3D.py
@@ -118,9 +118,9 @@ def generate_local_eulerian_grid_support_of_lagrangian_grid_kernel_3d(
 
         """
         # dtype of nearest_grid_index takes care of type casting to int
-        nearest_eul_grid_index_to_lag_grid[...] = (
-            lag_positions - eul_grid_coord_shift
-        ) // dx
+        nearest_eul_grid_index_to_lag_grid[...] = np.rint(
+            (lag_positions - eul_grid_coord_shift) / dx
+        )
         # TODO We need to add boundary exception handling! where the Lagrangian
         #  node goes in `interp_kernel_width` boundary zone of the Eulerian grid
         # get relative distance (support) of body

